### PR TITLE
docs+scripts+.github: add release notes template and a GH check

### LINKED ISSUE #765: please add human readable `docs/release-notes/release-notes-0.XX.X.md files`
Lightning Terminal releases just include auto generated release notes that are stored on GitHub from commit messages. Please add add human readable `docs/release-notes/release-notes-0.XX.X.md` files like LND has so that we can have a more human friendly understanding of major features and fixes made to lightning terminal that are archived in the git repository and can be automatically synced locally.

### PR DESCRIPTION
**Pull request by ellemouton**
Fixes #765

Here we add a template for 0.13.1-alpha release notes along with a github actions check to ensure that a PR always updates
the release notes as long as the (newly added) `no-changelog` label has not been added to the PR.

### CODE REVIEW

File: docs/release-notes/release-notes-0.13.1.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Release Notes
```
Comment (ViktorTigerstrom) [inline docs/release-notes/release-notes-0.13.1.md:1]: As I haven't really worked with the `lnd` release note flow yet, I'd just like to post a general question about the flow here so I understand the normal workflow during an actual release:

In the release PR, we'll remove all headers here which have no entry, is that correct?
Do we also add a new file for the next release in the release PR, or is that done in the first PR that gets merged?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -243,3 +243,19 @@ jobs:
...
+  ########################
+  release-notes-check:
+    name: check release notes updated
+    runs-on: ubuntu-latest
+    if: '!contains(github.event.pull_request.labels.*.name, ''no-changelog'')'
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (ViktorTigerstrom) [inline .github/workflows/main.yml:258]: Why do we need the `fetch-depth: 0` here, but not in lnd?

**Review by ViktorTigerstrom — COMMENTED** [review]
Looks great, thanks for adding this!

I've tested this locally as well by manipulating the `$GITHUB_REF` variable.

Just have some general questions before ACKing!

File: docs/release-notes/release-notes-0.13.1.md (hunk 1 above)
Comment (ellemouton) [reply docs/release-notes/release-notes-0.13.1.md:1]: > In the release PR, we'll remove all headers here which have no entry, is that correct?

yeah we can just remove what we dont need. Just added these for some structure. Not strict at all

Comment (ellemouton) [reply docs/release-notes/release-notes-0.13.1.md:1]: > Do we also add a new file for the next release in the release PR, or is that done in the first PR that gets merged?

Again not strict. I'd lean towards the later though

File: .github/workflows/main.yml (hunk 2 above)
Comment (ellemouton) [reply .github/workflows/main.yml:258]: i just used what is used else-where in lit

Comment (ellemouton) [reply .github/workflows/main.yml]: removed

**Review by ViktorTigerstrom — APPROVED** [review]
tACK LGTM :tada::fire::rocket!!

File: docs/release-notes/release-notes-0.13.1.md (hunk 1 above)
Comment (ViktorTigerstrom) [reply docs/release-notes/release-notes-0.13.1.md:1]: Ok cool, thanks for the clarification!

**Review by bitromortac — APPROVED** [review]
LGTM, nice!

**Pull request merged**